Using podman with 'make test' fails

I am using 'podman' and assumed I could run the unit tests with KPT_FN_RUNTIME="podman".

### Expected behavior

```KPT_FN_RUNTIME="podman" make test```

Assumed success

### Actual behavior

```
$ KPT_FN_RUNTIME="podman" make test
go test -cover ./...
?   	github.com/GoogleContainerTools/kpt	[no test files]
?   	github.com/GoogleContainerTools/kpt/commands	[no test files]
ok  	github.com/GoogleContainerTools/kpt/internal/builtins	(cached)	coverage: 87.1% of statements
ok  	github.com/GoogleContainerTools/kpt/internal/cmdapply	(cached)	coverage: 60.5% of statements
ok  	github.com/GoogleContainerTools/kpt/internal/cmddestroy	(cached)	coverage: 58.2% of statements
ok  	github.com/GoogleContainerTools/kpt/internal/cmddiff	(cached)	coverage: 85.0% of statements
Error: please ensure the container has an entrypoint and it supports --help flag: exec: "docker": executable file not found in $PATH
Usage:
  doc --image=IMAGE [flags]

Examples:

  # display the documentation for image set-namespace:v0.1.1
  kpt fn doc -i set-namespace:v0.1.1


Flags:
  -h, --help           help for doc
  -i, --image string   kpt function image name

--- FAIL: TestFnDoc (0.00s)
    cmdfndoc_test.go:52: 
        	Error Trace:	testutil.go:56
        	            				cmdfndoc_test.go:52
        	Error:      	Received unexpected error:
        	            	please ensure the container has an entrypoint and it supports --help flag: exec: "docker": executable file not found in $PATH
        	Test:       	TestFnDoc
        	Messages:   	[]
FAIL
coverage: 79.2% of statements
FAIL	github.com/GoogleContainerTools/kpt/internal/cmdfndoc	0.102s
```

### Information

Using 'main' branch

<!-- Please include any additional information related to your issue -->
<!-- EX: -->
<!--     Kpt Version (you may find this by running `kpt version` in your shell) -->
<!--     Kpt Package that can demonstrate the error. Ex: https://github.com/kubernetes/examples/staging/cockroachdb -->

Think the issue is [cmdfndoc assumes 'docker'](https://github.com/GoogleContainerTools/kpt/blob/main/internal/cmdfndoc/cmdfndoc.go#L76)





### Steps to reproduce the behavior

1. Assumes 'docker' is not installed on the machine (Tested with Linux and Mac)
2. Install podman
3. ```KPT_FN_RUNTIME="podman" make test```

--- a/internal/cmdfndoc/cmdfndoc.go
+++ b/internal/cmdfndoc/cmdfndoc.go
@@ -20,6 +20,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
 	"os/exec"
 
 	"github.com/GoogleContainerTools/kpt/internal/docs/generated/fndocs"
@@ -73,10 +74,21 @@ func (r *Runner) runE(c *cobra.Command, _ []string) error {
 		r.Image,
 		"--help",
 	}
-	cmd := exec.Command("docker", dockerRunArgs...)
+	// If the env var is empty, stringToContainerRuntime defaults it to docker.
+	runtime, err := fnruntime.StringToContainerRuntime(os.Getenv(fnruntime.ContainerRuntimeEnv))
+	if err != nil {
+		return err
+	}
+
+	err = fnruntime.ContainerRuntimeAvailable(runtime)
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command(runtime.GetBin(), dockerRunArgs...)
 	cmd.Stdout = &out
 	cmd.Stderr = &errout
-	err := cmd.Run()
+	err = cmd.Run()
 	pr := printer.FromContextOrDie(r.Ctx)
 	if err != nil {
 		pr.Printf(errout.String())

--- a/internal/fnruntime/container.go
+++ b/internal/fnruntime/container.go
@@ -101,18 +101,27 @@ type ContainerFn struct {
 	FnResult *fnresult.Result
 }
 
+func (r ContainerRuntime) GetBin() string {
+	switch r {
+	case Podman:
+		return podmanBin
+	default:
+		return dockerBin
+	}
+}
+
 // Run runs the container function using docker runtime.
 // It reads the input from the given reader and writes the output
 // to the provided writer.
 func (f *ContainerFn) Run(reader io.Reader, writer io.Writer) error {
 	// If the env var is empty, stringToContainerRuntime defaults it to docker.
-	runtime, err := stringToContainerRuntime(os.Getenv(ContainerRuntimeEnv))
+	runtime, err := StringToContainerRuntime(os.Getenv(ContainerRuntimeEnv))
 	if err != nil {
 		return err
 	}
 
 	checkContainerRuntimeOnce.Do(func() {
-		err = containerRuntimeAvailable(runtime)
+		err = ContainerRuntimeAvailable(runtime)
 	})
 	if err != nil {
 		return err
@@ -339,7 +348,7 @@ func isPodmanCLIoutput(s string) bool {
 	return false
 }
 
-func stringToContainerRuntime(v string) (ContainerRuntime, error) {
+func StringToContainerRuntime(v string) (ContainerRuntime, error) {
 	switch strings.ToLower(v) {
 	case string(Docker):
 		return Docker, nil
@@ -352,7 +361,7 @@ func stringToContainerRuntime(v string) (ContainerRuntime, error) {
 	}
 }
 
-func containerRuntimeAvailable(runtime ContainerRuntime) error {
+func ContainerRuntimeAvailable(runtime ContainerRuntime) error {
 	switch runtime {
 	case Docker:
 		return dockerCmdAvailable()

